pkg/metrics/metrics.go

```diff
@@ -394,6 +394,7 @@ func (c *Counter) ResetFailureMetrics() {
 		metricUpgradeControlPlaneTimeout,
 		metricUpgradeWorkerTimeout,
 		metricNodeDrainFailed,
+		metricUpgradeNotification,
```

Comment:
I think this will work, but we should ensure that this doesn't result in any unintended consequences such as MUO repeatedly trying to notify OCM that the upgrade has failed every reconcile until the UpgradeConfig eventually gets cleaned up.

My quick refresher of the code logic suggests to me that this shouldn't happen, but I'd still like to make sure that we test it and make sure.

Reply:
It should not I think, based on the controller logic, it should return directly when the upgradeconfig status is `Failed`
https://github.com/openshift/managed-upgrade-operator/blob/f38e833a50512e95c6238e79dac683f12302d7b2/controllers/upgradeconfig/upgradeconfig_controller.go#L227

And also during my testing, I cannot see it is trying to send the notification repeatedly.

Reply:
Thanks for confirming that it's been checked/tested, that's all I really wanted to know here.